runtime/xamarin/runtime.h

```diff
@@ -344,7 +344,7 @@ class XamarinObject {
 // Once we have one mono clone again the TARGET_OS_WATCH
 // condition should be removed (DYNAMIC_MONO_RUNTIME should still
 // be here though).
-#if TARGET_OS_WATCH && !defined (DYNAMIC_MONO_RUNTIME)
+#if (TARGET_OS_WATCH && !defined (DYNAMIC_MONO_RUNTIME)) || MONOMAC
```

Comment:
This can be simplified to:

```c
#if TARGET_OS_WATCH || MONOMAC
```

that said, is there a reason we should continue using the jit versions for iOS and tvOS? Are they faster than the coop versions, or do the coop versions not work correctly?

Reply:
The coop versions will work correctly with preemptive suspend.

There are a few reasons not to change iOS and tvOS:
1. Caution.  There's no particular reason motivating a change on those devices at this time.
2. Stack space usage: the coop functions need to store a cookie on the stack (in preemptive it'll always be junk) so there's a stack penalty for deeply recursive code that goes through many layers of managed->runtime->managed->native.  Not sure if any real code behaves like that, however.